### DISCUSSION TITLE
Usage of TMP_EXTERIOR_RAMP

### DISCUSSION THREAD

**Discussion by gesamarie** [firemodels/fds#13490] [Q&A] [answered]
Hi everyone,
I have a question regarding the usage of TMP_EXTERIOR_RAMP.

To summarize my Problem:
I have a house with an initial ambient temperature of 20°C ( which is set via TMPA=20.) 
I want to have different outside temperatures, which change over time.
I set the initial outside temperature via  
`&INIT XB=-0.3,6.1,0.3,2.3,-0.3,3.0, TEMPERATURE=8.0/ `
All mesh-edges, where the outside temperature should change, were implemented like this :
```
&VENT ID='Mesh Vent: main [ZMAX1]', SURF_ID='OPEN', XB=-2.3,-0.3,-5.6,2.3,3.0,3.0,TMP_EXTERIOR_RAMP=8.0, TMP_EXTERIOR_RAMP='tmp' /
&RAMP ID='tmp', T=0.0, F=1.0/
&RAMP ID='tmp', T=60, F=1.1/
```
**How exactly do you define the TMP_EXTERIOR_RAMP for TMP_EXTERIOR?**
I couldn't find any definition on how to define TMP_EXTERIOR_RAMP?
The results I am getting behave oppositional to what I would expect.
I did some tests to narrow down the behavior of my FDS Simulation:

Test 1: 
```
&RAMP ID='tmp', T=0.0, F=1.0/
&RAMP ID='tmp', T=60, F=0.9/
```
Test 2:
```
&RAMP ID='tmp', T=0.0, F=1.0/
&RAMP ID='tmp', T=60, F=1.0/
```
Test 3:
```
&RAMP ID='tmp', T=0.0, F=1.0/
&RAMP ID='tmp', T=60, F=1.1/
```
I would have expected, that the temperature outside (very close to the OPEN Vents) lowers for Test 1, stays the same for Test 2, and rises for Test 3.
My FDS Simulations did exactly the oposite.
After 60 seconds Test 1 had a temperature of 9.2°C, Test 2 stayed at +-8°C and Test 3 had a temperature of 6.8°C

If that would be helpful I could provide an exemplary Code.

Thank you in anvance for taking time to help me.

Best Regards,
Gesa

**Comment by mcgratta** [comment]
I'm surprised that the case ran with this line
```
&VENT ID='Mesh Vent: main [ZMAX1]', SURF_ID='OPEN', XB=-2.3,-0.3,-5.6,2.3,3.0,3.0,TMP_EXTERIOR_RAMP=8.0, TMP_EXTERIOR_RAMP='tmp' /
```

**Reply by gesamarie** [reply]
Oh Sorry, I copy and pasted wrong:
The line should look like this:
`&VENT ID='Mesh Vent: main [ZMAX1]', SURF_ID='OPEN', XB=-2.3,-0.3,-5.6,2.3,3.0,3.0,TMP_EXTERIOR=8.0, TMP_EXTERIOR_RAMP='tmp' / `

**Comment by mcgratta** [comment]
Could you post the input for Test 1 in the simplest possible way.

**Reply by gesamarie** [reply]
I hope this is sufficiently simple. I didn't change the geometry, just removed lots of unnecessary elements.
I can't upload .fds files -> just changed the .fds ending to the .txt ending.
[test_1_simple.txt](https://github.com/user-attachments/files/17151426/test_1_simple.txt)

**Reply by gesamarie** [reply]
My guess regarding the functionality of TMP_EXTERIOR_RAMP was as follows:

- define maximal TMP_EXTERIOR for the vent
- if I want to change the TMP_EXTERIOR of my vent within 60 seconds from TMP_EXTERIOR  to TMP_EXTERIOR/2 I would define the RAMP like this:
```
&RAMP ID='tmp', T=0.0, F=1.0/
&RAMP ID='tmp', T=60, F=0.5/
```
Is this a correct definition of RAMP ?

**Comment by mcgratta** [comment, marked as answer]
$T(t) = T_\infty + f(T_{\rm s} - T_\infty)$

$T_\infty = 20$ C
$T_{\rm s} = 8$ C
$f=0.9$

I would expect the final outside temperature to be approximately 9.2 C. This appears to be what I am seeing.

**Reply by gesamarie** [reply]
I didn't know, that the TMP_EXTERIOR_RAMP was defined in that way.
I tried to find a definition in the UserGuide before, but failed.
I looked again this morning and found the definition (11.2), with the knowledge how the function is supposed to look.
With this T(t) function the RAMP behaves as expected.

Thank you for your help.

**Comment by mcgratta** [comment]
I'll add a link to the chapter on ramps. All temperature ramps work this way, but we'd prefer to explain it in one place.

**Reply by gesamarie** [reply]
That makes total sense not to define severall times, the link is a good idea :)
